frontend/src/app/main/ui/ds/controls/utilities/input_field.cljs

```diff
@@ -0,0 +1,72 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.main.ui.ds.controls.utilities.input-field
+  (:require-macros
+   [app.common.data.macros :as dm]
+   [app.main.style :as stl])
+  (:require
+   [app.common.data :as d]
+   [app.main.constants :refer [max-input-length]]
+   [app.main.ui.ds.foundations.assets.icon :refer [icon* icon-list]]
+   [app.util.dom :as dom]
+   [rumext.v2 :as mf]))
+
+(def ^:private schema:input-field
+  [:map
+   [:class {:optional true} :string]
+   [:id :string]
+   [:icon {:optional true}
+    [:and :string [:fn #(contains? icon-list %)]]]
+   [:has-hint {:optional true} :boolean]
+   [:hint-type {:optional true} [:maybe [:enum "hint" "error" "warning"]]]
+   [:type {:optional true} :string]
+   [:max-length {:optional true} :int]
+   [:variant {:optional true} [:enum "seamless" "dense" "comfortable"]]])
+
+(mf/defc input-field*
+  {::mf/props :obj
+   ::mf/forward-ref true
+   ::mf/schema schema:input-field}
+  [{:keys [id icon has-hint hint-type class type max-length variant children] :rest props} ref]
+  (let [ref   (or ref (mf/use-ref))
+        type  (d/nilv type "text")
+        variant (d/nilv variant "dense")
+        props (mf/spread-props props
+                               {:class (stl/css-case
+                                        :input true
+                                        :input-with-icon (some? icon))
+                                :ref ref
+                                :aria-invalid (when (and has-hint
+                                                         (= hint-type "error"))
+                                                "true")
+                                :aria-describedby (when has-hint
+                                                    (str id "-hint"))
+                                :type (d/nilv type "text")
+                                :id id
+                                :maxLength (d/nilv max-length (str max-input-length))})
```

Comment:
Should be `:max-lenght` for case consistency. It will be converted to `maxLength` automatically